src/js/core/UpdateStream.ts

```diff
@@ -0,0 +1,158 @@
+    versionObject.response &&
+    versionObject.response.results
+  ) {
+    return Object.keys(versionObject.response.results)[0];
+  }
+  return "0";
+}
+
+function getFromLocalStorage(key: string): string | number | null {
+  const value: any = Util.findNestedPropertyInObject(
+    UserSettingsStore.getKey(SAVED_STATE_KEY),
+    DCOS_UI_VERSION_SETTING + "." + key
+  );
+
+  return value;
+}
+
+function setInLocalStorage(key: string, value: any): void {
+  const savedStates: DCOSUserSettings =
+    UserSettingsStore.getKey(SAVED_STATE_KEY) || {};
+  savedStates.dcosUIVersion = {
+    dismissedVersion: localStorageDismissedVersion.getValue(),
+    lastTimeCheck: localStorageCheckedTime.getValue()
+  };
+  savedStates.dcosUIVersion[key] = value;
+  UserSettingsStore.setKey(SAVED_STATE_KEY, savedStates);
+  if (key === DISMISSED_VERSION) {
+    localStorageDismissedVersion.next(value);
+  } else if (key === LAST_TIME_CHECK) {
+    localStorageCheckedTime.next(value);
+  }
+}
+
+function getCosmosHeaderString({
+  action,
+  actionType,
+  entity,
+  version
+}: ContentTypeObject): string {
+  return `application/vnd.dcos.${entity}.${action}-${actionType}+json;charset=utf-8;version=${version}`;
+}
+
+const localStorageDismissedVersion: BehaviorSubject<
+  string | number | null
+> = new BehaviorSubject(
+  getFromLocalStorage(DISMISSED_VERSION)
+    ? getFromLocalStorage(DISMISSED_VERSION)
+    : null
+);
+const localStorageCheckedTime: BehaviorSubject<
+  string | number | null
+> = new BehaviorSubject(
+  getFromLocalStorage(LAST_TIME_CHECK)
+    ? getFromLocalStorage(LAST_TIME_CHECK)
+    : null
+);
+
+const filteredDismissedVersion: Observable<
+  string | number | null
+> = localStorageDismissedVersion.filter(
+  () =>
+    new Date().getTime() >=
+    Number(localStorageCheckedTime.getValue()) + CHECK_DELAY
+);
+
+const fetchedVersion: Observable<RequestResponse<ResponseObject>> = request(
+  "/package/list-versions",
+  {
+    method: "POST",
+    headers: {
+      "Content-Type": getCosmosHeaderString({
+        action: "list-versions",
+        actionType: "request",
+        entity: "package",
+        version: "v1"
+      }),
+      Accept: getCosmosHeaderString({
+        action: "list-versions",
+        actionType: "response",
+        entity: "package",
+        version: "v1"
+      })
+    },
+    body: JSON.stringify({
+      includePackageVersions: true,
+      packageName: "dcos-ui"
+    })
+  }
+).retry(4);
+
+export function compareStream(
+  delay: number,
+  dismissedVersion: Observable<string | number | null>,
+  newFetchedVersion: Observable<VersionObject>
+): Observable<string | number | null> {
+  return Observable.timer(0, delay)
+    .switchMap(() =>
+      Observable.combineLatest(dismissedVersion, newFetchedVersion)
+    )
+    .filter(([dismissed, nextVersion]) => {
+      const newVersion = getVersionFromVersionObject(nextVersion);
+
+      return compareVersions(newVersion, String(dismissed)) === 1;
```

Comment:
If the dismissed version is `null` (which is currently allowed by our types), this errors with `Uncaught Error: Invalid argument not valid semver ('null' received)`. I think we either need to enforce that this is not null in the typing or do something like `return dismissed ? compareVersions(newVersion, String(dismissed)) === 1 : true;`

Reply:
Also even if the dismissed version is a string but `not valid semver` this will error

Reply:
You are right. When I wrote it like that I was thinking that `String(null)` is `"null"`, but I didn't consider that `comapreVersions` expect a specific string.